internal/database/repos_perm.go

```diff
@@ -71,56 +75,43 @@ OR (
 		AND unrestricted
 	)
 )
-OR  (
-	NOT %s                        -- Disregard unrestricted state when permissions user mapping is enabled
-	AND (
-		NOT repo.private          -- Happy path of non-private repositories
-		OR  EXISTS (              -- Each external service defines if repositories are unrestricted
-			SELECT
-			FROM external_services AS es
-			JOIN external_service_repos AS esr ON (
-					esr.external_service_id = es.id
-				AND esr.repo_id = repo.id
-				AND es.unrestricted = TRUE
-				AND es.deleted_at IS NULL
-			)
-		)
-	)
-)
-OR  (                             -- Restricted repositories require checking permissions
-	(
-		SELECT object_ids_ints @> INTSET(repo.id)
-		FROM user_permissions
-		WHERE
-			user_id = %s
-		AND permission = %s
-		AND object_type = 'repos'
-	) AND EXISTS (
-		SELECT
-		FROM external_service_repos
-		WHERE repo_id = repo.id
-		AND (
-				(user_id IS NULL AND org_id IS NULL)  -- The repository was added at the instance level
-			OR  user_id = %s                          -- The authenticated user added this repository
-			OR  EXISTS (                              -- The authenticated user is a member of an organization that added this repository
-				SELECT
-				FROM org_members
-				WHERE
-					external_service_repos.org_id = org_members.org_id
-				AND org_members.user_id = %s
-			)
-		)
+`
+	const externalServiceUnrestrictedQuery = `
+OR (
+    NOT repo.private          -- Happy path of non-private repositories
+    OR  EXISTS (              -- Each external service defines if repositories are unrestricted
+        SELECT
+        FROM external_services AS es
+        JOIN external_service_repos AS esr ON (
+                esr.external_service_id = es.id
+            AND esr.repo_id = repo.id
+            AND es.unrestricted = TRUE
+            AND es.deleted_at IS NULL
+        )
 	)
 )
+`
+	const restrictedRepositoriesQuery = `
+OR (                             -- Restricted repositories require checking permissions
+    SELECT object_ids_ints @> INTSET(repo.id)
+    FROM user_permissions
+    WHERE
+        user_id = %s
+    AND permission = %s
+    AND object_type = 'repos'
 )
 `
+	queryFmtString := "(" + unrestrictedReposQuery
+
+	// Disregard unrestricted state when permissions user mapping is enabled
+	if !usePermissionsUserMapping {
+		queryFmtString = queryFmtString + externalServiceUnrestrictedQuery
+	}
+
+	queryFmtString = queryFmtString + restrictedRepositoriesQuery + ")"
```

Comment:
Should we even look at `restrictedRepositoriesQuery` if `usePermissionsUserMapping` is `false`?

Just by eyeballing this PR, I would expect the following code instead:
```suggestion
    conditionalQuery := restrictedRepositoriesQuery
	// Disregard unrestricted state when permissions user mapping is enabled
	if !usePermissionsUserMapping {
		conditionalQuery = externalServiceUnrestrictedQuery
	}
	queryFmtString := "(" + unrestrictedReposQuery + conditionalQuery + ")"
```

Reply:
The original code still checked restricted repositories if usePermissionsUserMapping was false 🤔 

Not super sure here